src/test/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifierTestInt.java

```diff
@@ -0,0 +1,140 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Matchers;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+/**
+ * @author Damian Szczepanik (damianszczepanik@github)
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({DisplayURLProvider.class, Office365ConnectorWebhookNotifier.class, Run.class})
+public class Office365ConnectorWebhookNotifierTestInt {
+
+    public static final String JOB_NAME = "myFirstJob";
+    private static final int BUILD_NUMBER = 167;
+    private static final long START_TIME = 1508617305000L;
+
+    private AbstractBuild run;
+    private HttpWorkerAnswer workerAnswer;
+
+    @Before
+    public void setUp() {
+        mockListener();
+
+        run = mockRun();
+        when(run.getChangeSet()).thenReturn(new ChangeLogSetBuilder(run));
+
+        mockDisplayURLProvider();
+        mockEnvironment();
+        mockHttpWorker();
+    }
+
+    private AbstractBuild mockRun() {
+        AbstractBuild run = mock(AbstractBuild.class);
+
+        when(run.getNumber()).thenReturn(BUILD_NUMBER);
+        PowerMockito.when(run.getStartTimeInMillis()).thenReturn(START_TIME);
+
+        Job job = mockJob();
+        when(run.getParent()).thenReturn(job);
+
+        // getProperty
+        WebhookJobProperty property = new WebhookJobProperty(WebhookBuilder.sampleWebhookWithAllStatuses());
+        when(job.getProperty(WebhookJobProperty.class)).thenReturn(property);
+
+        return run;
+    }
+
+    private Job mockJob() {
+        Job job = mock(Job.class);
+
+        when(job.getDisplayName()).thenReturn(JOB_NAME);
+
+        return job;
+    }
+
+    private TaskListener mockListener() {
+        TaskListener listener = mock(TaskListener.class);
+
+        PrintStream stream = mock(PrintStream.class);
+        when(listener.getLogger()).thenReturn(stream);
+
+        return listener;
+    }
+
+    private void mockDisplayURLProvider() {
+        mockStatic(DisplayURLProvider.class);
+        when(DisplayURLProvider.get()).thenReturn(new ClassicDisplayURLProviderBuilder());
+    }
+
+    private void mockEnvironment() {
+        EnvVars envVars = mock(EnvVars.class);
+        try {
+            TaskListener taskListener = Matchers.any();
+            when(run.getEnvironment(taskListener)).thenReturn(envVars);
+        } catch (IOException | InterruptedException e) {
+            throw new IllegalArgumentException(e);
+        }
+        when(envVars.expand(ClassicDisplayURLProviderBuilder.URL)).thenReturn(ClassicDisplayURLProviderBuilder.URL);
+    }
+
+    private void mockHttpWorker() {
+        workerAnswer = new HttpWorkerAnswer();
+        try {
+            whenNew(HttpWorker.class).withAnyArguments().thenAnswer(workerAnswer);
+        } catch (Exception e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
```

Comment:
I did not mock everything but only a few classes so review process can be easier.

The goal is to merge skeleton and then add more tests